recipes/snappy/all/conanfile.py

```diff
@@ -53,8 +57,8 @@ def generate(self):
         tc.variables["SNAPPY_BUILD_TESTS"] = False
         if Version(self.version) >= "1.1.8":
             tc.variables["SNAPPY_FUZZING_BUILD"] = False
-            tc.variables["SNAPPY_REQUIRE_AVX"] = False
-            tc.variables["SNAPPY_REQUIRE_AVX2"] = False
+            tc.variables["SNAPPY_REQUIRE_AVX"] = self.options.get_safe("with_avx") in ("avx", "avx2")
+            tc.variables["SNAPPY_REQUIRE_AVX2"] = self.options.get_safe("with_avx") == "avx2"
```

Comment:
Looking at how it is used upstream, these options make little sense for conan since it's just a babysitting of compiler flags without more logic. This is why I've never added these options in previous snappy PR.